src/grid/asGrid.js

```diff
@@ -52,8 +52,8 @@ export default function asGrid(Component: *) {
   ) {
     const classes = compact([
       className,
-      size && styles.col,
-      size && styles[`col-${String(size)}`],
+      size !== undefined && styles.col,
+      size !== undefined && styles[`col-${String(size)}`],
```

Comment:
here we don't have the same guarantees on undefined, if we want to check for it we probably want to do `typeof size !== 'undefined'`